src/config/Configuration.h

```diff
@@ -8,15 +8,16 @@
 using json = nlohmann::json;
 
 namespace config {
-class Configuration : public ConfigurationInterface<json> {
```

Comment:
This way we'll have to implement a class for every format (XML, YAML, ini etc.) right? If yes, maybe it could be more clear renaming it to something like ConfigurationJSON. If no, maybe we can simply overload the load, write and create default methods

Reply:
Yes, you're right, better renaming it and if we come across another type of file, we'll create another class.